src/modules/authentacation/components/Register/Register.tsx

```diff
@@ -1,7 +1,248 @@
+    const formData = new FormData();
+    Object.entries(data).forEach(([key, value]) => {
+      if (key === 'profileImage' && value instanceof File) {
+        formData.append(key, value); // Handle file specifically
+      } else {
+        formData.append(key, value as string);
+      }
+    });
+
+    try {
+      const response = await publicAxiosInstance.post(
+        USERS_URLS.REGISTER,
+        formData,
+        {
+          headers: {
+            'Content-Type': 'multipart/form-data',
+          },
+        }
+      );
+    
+      toast.success(response?.data?.message || 'Register Successfully');
+      navigate('/Verify', { state: data.email });
+    } catch (error) {
+      console.error('Registration error:', error);
+      const errorMessage =
+        error?.response?.data?.message || 'Registration failed. Please try again.';
+      toast.error(errorMessage);
+    }
+  }; // <-- This was missing
+
   return (
-    <div>Register</div>
-  )
+    <div className="w-full">
+      <div className="my-6">
+        <h6 className="text-white text-sm">Welcome to PMS</h6>
+        <h2 className="text-primary font-bold text-3xl tracking-wide">
+          <span className="capitalize border-b-4 border-primary">R</span>egister
+        </h2>
+      </div>
+
+      <form onSubmit={handleSubmit(onSubmit)} className="w-full">
+        <div className="flex justify-center flex-col items-center mb-8 relative">
+          <label htmlFor="profileImage" className="cursor-pointer relative ">
+            <img
+              src={imagePreview || human}
+              alt="Profile"
+              className={` w-[70px] h-[70px] rounded-full border border-gray-300 bg-[#3159518A]   ${isImageSelected ? 'opacity-80' : ''}`} // Apply opacity when image is selected
+            />
+            <FaCamera className="absolute top-[26px] right-[26px] text-white text-xl opacity-50  " /> 
+          </label>
+          <input
+            type="file"
+            id="profileImage"
+            ref={fileInputRef}
+            className="hidden"
+            {...register('profileImage')}
+            onChange={handleFileChange}
+            accept="image/*"
+          />
+        </div>
+
+        <div className="grid grid-cols-2 gap-6">
+          <div>
+            <label className="block text-primary">User Name</label>
+            <input
+              type="text"
+              placeholder="Name"
+              className="bg-inherit placeholder-gray-300 pb-2 border-b border-gray-400 w-full placeholder:tracking-wide focus:outline-none text-white"
+              {...register('userName', { required: 'User name is required' })}
+            />
+            {errors?.userName && (
+              <span className="text-red-500 text-sm">
+                {errors?.userName?.message}
+              </span>
+            )}
+          </div>
+
+          <div>
+            <label className="block text-primary">E-mail</label>
+            <input
+              type="email"
```

Comment:
This will prevent the react-hook-form validation while submitting if the email is incorrect, as it will trigger the html validation for email.